### PR TITLE
Add API index

### LINKED ISSUE #427: Add nicer html page for https://hex.pm/api
Currently it 404s, perhaps we could show a simple html page with links to API docs or something like that.

### PR DESCRIPTION
**Pull request by wojtekmach**
Closes #427

### CODE REVIEW

File: web/views/api/index_view.ex [hunk 1]
```diff
@@ -0,0 +1,8 @@
+defmodule HexWeb.API.IndexView do
+  use HexWeb.Web, :view
+
+  def render("index." <> _format, _assigns) do
+    %{packages_url: api_package_url(Endpoint, :index),
+      documentation_url: "http://docs.hexpm.apiary.io"}
+  end
```
Comment (wojtekmach) [inline web/views/api/index_view.ex]: @ericmj we could also add "GitHub API-style" placeholder urls (api.github.com) so this whole response would look like this:

```
{
  "packages_url":        "http://localhost:4000/api/packages",
  "package_url":         "http://localhost:4000/api/packages/{name}",
  "package_release_url": "http://localhost:4000/api/packages/{name}/releases/{version}",
  "package_owners_url":  "http://localhost:4000/api/packages/{name}/owners",
  "user_url":            "http://localhost:4000/api/users/{name}",
  "documentation_url":   "http://docs.hexpm.apiary.io"
}
```

Comment (wojtekmach) [inline web/views/api/index_view.ex]: oh, there's also `/api/keys` endpoint which we could make discoverable here, but it requires key auth (but prompts for username/password http basic auth when accessed from the browser which won't authenticate)

Comment (ericmj) [inline web/views/api/index_view.ex]: That's a great idea. I was thinking about how to make the non-index resources discoverable.

We should add keys even though they require auth.

**Comment by wojtekmach** [comment]
@ericmj updated

**Pull request merged**